File: frontend/src/api/health.ts
// AI Optimized by Skills Agent: Health API contract validates frontend-backend connectivity.
import { apiGet } from "./client";

export type HealthResponse = {
  status: string;
  service: string;
  version: string;
};

// AI Optimized by Skills Agent: Small typed request used by the app shell smoke state.
export function getHealth() {
  return apiGet<HealthResponse>("/health");
}

File: backend/src/app/api/routes/health.py
# AI Optimized by Skills Agent: Minimal health route for skeleton verification and CI smoke tests.
from fastapi import APIRouter

from app.schemas.health import HealthResponse

router = APIRouter(tags=["health"])


# AI Optimized by Skills Agent: Returns static service metadata without touching external systems.
@router.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    return HealthResponse(status="ok", service="quick-career-api", version="0.1.0")
